src/material-luxon-adapter/adapter/luxon-date-adapter.ts

```diff
@@ -0,0 +1,254 @@
+    return date.weekday;
+  }
+
+  getMonthNames(style: 'long' | 'short' | 'narrow'): string[] {
+    return Info.months(style, {locale: this.locale});
+  }
+
+  getDateNames(): string[] {
+    // At the time of writing, Luxon doesn't offer similar
+    // functionality so we have to fall back to the Intl API.
+    const dtf = new Intl.DateTimeFormat(this.locale, {day: 'numeric', timeZone: 'utc'});
+
+    return range(31, i => {
+      // Format a UTC date in order to avoid DST issues.
+      const date = DateTime.utc(2017, 1, i + 1).toJSDate();
+
+      // Strip the directionality characters from the formatted date.
+      return dtf.format(date).replace(/[\u200e\u200f]/g, '');
+    });
+  }
+
+  getDayOfWeekNames(style: 'long' | 'short' | 'narrow'): string[] {
+    // Note that we shift the array once, because Luxon returns Monday as the
+    // first day of the week, whereas our logic assumes that it's Sunday. See:
+    // https://moment.github.io/luxon/docs/class/src/info.js~Info.html#static-method-weekdays
+    const days = Info.weekdays(style, {locale: this.locale});
+    days.unshift(days.pop()!);
+    return days;
+  }
+
+  getYearName(date: DateTime): string {
+    return date.toFormat('yyyy');
+  }
+
+  getFirstDayOfWeek(): number {
+    // Luxon doesn't have support for getting the first day of the week.
+    return 0;
+  }
+
+  getNumDaysInMonth(date: DateTime): number {
+    return date.daysInMonth;
+  }
+
+  clone(date: DateTime): DateTime {
+    return DateTime.fromObject(date.toObject());
+  }
+
+  createDate(year: number, month: number, date: number): DateTime {
+    if (month < 0 || month > 11) {
+      throw Error(`Invalid month index "${month}". Month index has to be between 0 and 11.`);
+    }
+
+    if (date < 1) {
+      throw Error(`Invalid date "${date}". Date has to be greater than 0.`);
+    }
+
+    // Luxon uses 1-indexed months so we need to add one to the month.
+    const result =
+        this._useUTC ? DateTime.utc(year, month + 1, date) : DateTime.local(year, month + 1, date);
+
+    if (!this.isValid(result)) {
+      throw Error(`Invalid date "${date}". Reason: "${result.invalidReason}".`);
+    }
+
+    return result.setLocale(this.locale);
+  }
+
+  today(): DateTime {
+    return (this._useUTC ? DateTime.utc() : DateTime.local()).setLocale(this.locale);
+  }
+
+  parse(value: any, parseFormat: string | string[]): DateTime | null {
+    const options: DateTimeOptions = this._getOptions();
+
+    if (typeof value == 'string' && value.length > 0) {
+      const iso8601Date = DateTime.fromISO(value, options);
+
+      if (this.isValid(iso8601Date)) {
+        return iso8601Date;
+      }
+
+      let formats: string[];
+
+      if (Array.isArray(parseFormat)) {
+        if (!parseFormat.length) {
+          throw Error('Formats array must not be empty.');
+        }
+        formats = parseFormat;
+      } else {
+        formats = [parseFormat];
+      }
+
+      for (let i = 0; i < formats.length; i++) {
```

Comment:
```suggestion
      for (const format of formats) {
```
Looks like you don't need the index?